strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/proxied/LocalStorageProxyRepositoryArtifactResolver.java

```diff
@@ -112,6 +112,7 @@ protected InputStream onSuccessfulProxyRepositoryResponse(InputStream is,
         try (// Wrap the InputStream, so we could have checksums to compare
              final InputStream remoteIs = new MultipleDigestInputStream(is))
         {
+            //XXX: is this proper implementation?
             long totalNumberOfArtifactBytes = layoutProvider.getArtifactManagementService().store(tempArtifact,
```

Comment:
@fuss86 can you please have a look at this?
If I understand correctly we will never create `RemoteArtifactEntry` as it supposed to be here.

Reply:
Also there must be `ClassCastException` [here](https://github.com/strongbox/strongbox/pull/424/files#diff-3b9c95f0105d7ef603a26f0811c37dd0L122)

Reply:
As we talked @sbespalov , please take a look at `tempArtifact` and `TODO` over `org.carlspring.strongbox.services.impl.AbstractArtifactManagementService#storeArtifactEntry` method